android/src/com/mapswithme/maps/gallery/FullScreenGalleryActivity.java

```diff
@@ -0,0 +1,198 @@
+      viewPager.setAdapter(mGalleryPageAdapter);
+      viewPager.setCurrentItem(mPosition);
+      viewPager.post(new Runnable()
+      {
+        @Override
+        public void run()
+        {
+          onPageSelected(viewPager.getCurrentItem());
+        }
+      });
+    }
+  }
+
+  @Override
+  public int getThemeResourceId(String theme)
+  {
+    if (ThemeUtils.isDefaultTheme(theme))
+      return R.style.MwmTheme_FullScreenGalleryActivity;
+
+    if (ThemeUtils.isNightTheme(theme))
+      return R.style.MwmTheme_Night_FullScreenGalleryActivity;
+
+    throw new IllegalArgumentException("Attempt to apply unsupported theme: " + theme);
+  }
+
+  @Override
+  protected int getContentLayoutResId()
+  {
+    return R.layout.activity_full_screen_gallery;
+  }
+
+  @Override
+  public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels)
+  {
+  }
+
+  @Override
+  public void onPageSelected(int position)
+  {
+    updateInformation(mGalleryPageAdapter.getImage(position));
+  }
+
+  @Override
+  public void onPageScrollStateChanged(int state)
+  {
+  }
+
+  private void readParameters()
+  {
+    Bundle extras = getIntent().getExtras();
+    if (extras != null)
+    {
+      mImages = extras.getParcelableArrayList(EXTRA_IMAGES);
+      mPosition = extras.getInt(EXTRA_POSITION);
+    }
+  }
+
+  private void updateInformation(@NonNull Image image)
+  {
+    UiUtils.setTextAndHideIfEmpty(mDescription, image.getDescription());
+    UiUtils.setTextAndHideIfEmpty(mUserName, image.getUserName());
+    UiUtils.setTextAndHideIfEmpty(mSource, image.getSource());
+    updateDate(image);
+    updateUserAvatar(image);
+    updateUserBlock();
+  }
+
+  private void updateDate(Image image)
+  {
+    if (image.getDate() != null)
+    {
+      Date date = new Date(image.getDate());
+      mDate.setText(DateFormat.getMediumDateFormat(this).format(date));
+      UiUtils.show(mDate);
+    }
+    else
+      UiUtils.hide(mDate);
+  }
+
+  private void updateUserAvatar(Image image)
+  {
+    if (!TextUtils.isEmpty(image.getUserAvatar()))
+    {
+      UiUtils.show(mAvatar);
+      Glide.with(this)
+           .load(image.getUserAvatar())
+           .asBitmap()
+           .centerCrop()
+           .into(new BitmapImageViewTarget(mAvatar)
+           {
+             @Override
+             protected void setResource(Bitmap resource)
+             {
+               RoundedBitmapDrawable circularBitmapDrawable =
+                   RoundedBitmapDrawableFactory.create(getResources(), resource);
+               circularBitmapDrawable.setCircular(true);
+               mAvatar.setImageDrawable(circularBitmapDrawable);
+             }
+           });
+    }
+    else
+      UiUtils.hide(mAvatar);
+  }
+
+  private void updateUserBlock()
+  {
+    if (UiUtils.isHidden(mUserName)
+        && UiUtils.isHidden(mSource)
+        && UiUtils.isHidden(mDate)
+        && UiUtils.isHidden(mAvatar))
+      UiUtils.hide(mUserBlock);
```

Comment:
Please, add curly braces around positive branch and negative branch, i.e.:
```java
if (...)
{
  UiUtils.hide(mUserBlock);
}
else
{
  UiUtils.show(mUserBlock);
}
```
Otherwise it's hard to read this statement.